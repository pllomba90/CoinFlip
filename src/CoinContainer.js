import React, { useState } from "react";
import Images from "./images";
import pick from "./helpers";
import Coin from "./Coin";

const CoinContainer = (props)=>{

    const [coin, setCoin] = useState(null);
    const [headCount, setHeadCount] = useState(0);
    const [tailCount, setTailCount] = useState(0);

    const handleClick = () => {
        const newCoin =pick(Images);
        setCoin(newCoin);
        if (newCoin.alt === "heads") {
          setHeadCount(oldCount => oldCount + 1);
        } else {
          setTailCount(oldCount => oldCount + 1);
        }
      };

      const currCoin = coin ? (
        <Coin side={coin.alt} src={coin.src} />
      ) : null;
    return (
        <div className="coincontainer">
            <h1>Flip a Coin!</h1>
            {currCoin}
            <button  onClick={handleClick}>Flip!</button>
            <p>Out of {headCount + tailCount} flips, there have been {headCount} heads
        and {tailCount} tails.</p>
        </div>
    );
}


export default CoinContainer;




