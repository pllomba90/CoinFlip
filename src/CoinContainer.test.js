import React from "react";
import { render } from "@testing-library/react";
import CoinContainer from "./CoinContainer";

it("renders without crashing", function() {
  render(<CoinContainer />);
});

it("matches snapshot", function() {
    const {asFragment} = render(<CoinContainer />);
    expect(asFragment()).toMatchSnapshot();
  });