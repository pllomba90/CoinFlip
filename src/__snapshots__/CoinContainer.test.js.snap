// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="coincontainer"
  >
    <h1>
      Flip a Coin!
    </h1>
    <button>
      Flip!
    </button>
    <p>
      Out of 0 flips, there have been 0 heads and 0 tails.
    </p>
  </div>
</DocumentFragment>
`;
